File: Modified Numbers and Queries - GFG - POTD - 03_10_2022/Modified_Numbers_and_Queries.cpp
// { Driver Code Starts 
// Initial Template for C++ 
#include <bits/stdc++.h>
using namespace std; 
// }
// User Function Template for C++. 

class solution {
    public:
        int sumOfAll(int l , int r)
        {
            // code here
            vector<int> sieve(r + 1) ; 
            vector<int> dp(r + 1) ; 
            dp[1] = 1 ; 

            for(int i = 2 ; i <= r ; i++)
            {
                if(sieve[i] == 0)
                {
                    for(int j = i ; j <= r ; j += 1)
                    {
                        sieve[j] = 1 ; 
                        dp[j] += i ; 

                    }
                }
            }
            return accumulate(dp.begin() + l , dp.end() , 0) ; 

        }
}; 


// { Driver Code Starts 
int main()
{
    int t ; 
    cin >> t ; 
    while(t--)
    {
        int l , r ; 
        cin >> l >> r ; 
        solution ob ; 
        cout << ob.sumOfAll(l, r) << endl ; 
    }
    return 0 ;
}

// } Driver Code Ends

File: GFG/Modified Numbers and Queries/Modified Numbers and Queries.py
class Solution:
	def sumOfAll(self, l, r):
		# code here
        arr=[0]*(r+1)
        arr[1]=1
        
        for i in range(2,r+1):
            if arr[i]!=0:
                continue
        
            for j in range(i,r+1,i):
                if j%i==0:
                    arr[j]+=i
        
        count=0
        
        for i in range(l,r+1):
            count+=arr[i]
        
        return count

#{ 
 # Driver Code Starts
#Initial Template for Python 3
if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		l,r = input().split()
		l=int(l)
		r=int(r)
		ob = Solution();
		print(ob.sumOfAll(l,r))

# } Driver Code Ends
